subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationSpec.groovy

```diff
@@ -94,6 +105,18 @@ import static org.hamcrest.CoreMatchers.equalTo
 import static org.hamcrest.MatcherAssert.assertThat
 
 class DefaultConfigurationSpec extends Specification implements InspectableConfigurationFixture {
+    final OutputEventListener outputEventListener = new OutputEventListener() {
+        final StringBuilder output = new StringBuilder()
+
+        @Override
+        void onOutput(OutputEvent event) {
+            LogEvent logEvent = event as LogEvent
+            output.append(logEvent.message)
+        }
+    }
+    @Rule
+    final ConfigureLogging logging = new ConfigureLogging(outputEventListener)
```

Comment:
Why add this?

Reply:
It could be removed, but it took me a minute to figure out how to successfully test for the new deprecations here (in the original branch where I deprecated and logged), so I figured it might be useful to leave in for forthcoming deprecations to be made to Configuration related stuff in 8.3.

Reply:
This discovery could easily be preserved in a branch and doesn't need to be merged into master. 

Also don't we traditionally test deprecations in integration tests anyways?

Reply:
Sure, I'll remove it until we need it.

The deprecation warning seemed relevant to test in this case, considering how this test uses `ResolvedConfiguration`.